### PR TITLE
Lerna working builds

### PR DESCRIPTION
**Pull request by nealfennimore**
- Adds in lerna for building between projects
- Abstract out webpack configs for reuse
- Update some [plugins](https://github.com/moderntribe/events-gutenberg/commit/4158517000d6c9857498ab4ed71df8f9fab53cdb)
- Add in webpack and tribe common plugin to global webpack configs
- Add in analyze, dev, and build scripts

### CODE REVIEW

File: webpack/utils/directories.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { lstatSync, readdirSync } = require( 'fs' );
+const { join } = require( 'path' );
+
+const isDirectory = source => lstatSync( source ).isDirectory();
+const getDirectories = source => (
```
Comment (mitogh) [inline webpack/utils/directories.js]: Wonder why not inline export instead. `export const ...`

File: webpack/module/rules/styles.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
+
+const plugin = new MiniCssExtractPlugin( {
```
Comment (mitogh) [inline webpack/module/rules/styles.js]: Same question about the inline export.

File: webpack/externals/tribe/common.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const pickBy = require( 'lodash/pickBy' );
```
Comment (mitogh) [inline webpack/externals/tribe/common.js]: How about `import { pickBy } form 'lodash'`

File: webpack/utils/directories.js (hunk 1 above)
Comment (nealfennimore) [reply webpack/utils/directories.js]: No this needs to be in `commonjs` syntax - we're not compiling ES6 modules down to CJS syntax before webpack can use it

Comment (mitogh) [reply webpack/utils/directories.js]: Oh! There was a reason 😆 .

Thanks for the heads up.

File: webpack/externals/tribe/common.js (hunk 3 above)
Comment (nealfennimore) [reply webpack/externals/tribe/common.js]: Not necessary in this case, lodash is already modularized and this is part of the build process

**Comment by nealfennimore** [comment]
@bordoni @mitogh I made a couple changes:
- Updated to use `npm` instead of `yarn`... `yarn` was causing issues with workspaces  and how that was working in conjunction with lerna.
- `@@tribe` is now `@moderntribe`
- `window.tribe.common.hoc` now exists and is properly bundled away from `window.tribe.events`. I now need this file `hoc.js` in the asset pipeline as the events plugin depends on it. 
- `common` plugin has some dependencies on `events` plugin and that can be removed now
- Added in an `analyze` script so you guys can visually see the bundle dependencies (we should probably add a vendor bundle, there's a lot of duplicate vendor modules)

**Comment by bordoni** [comment]
I like those changes, my only problem is, do we need to release stuff into NPM? For `@moderntribe`?

**Pull request merged**